app/javascript/react/components/ParkNewContainer.js

```diff
@@ -1,13 +1,46 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { Redirect } from 'react-router-dom'
 
 import ParkNewForm from './ParkNewForm'
 
 const ParkNewContainer = (props) => {
+  const [shouldShowForm, setShouldShowForm] = useState(false)
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+
+  useEffect(() => {
+    fetch("/api/v1/users/get_current_user")
+    .then(response => {
+      if(response.ok) {
+        return response.json()
+      } else {
+        throw new Error(`${response.status}: ${response.statusText}`)
+      }
+    })
+    .then(parsedBody => {
+      if(parsedBody.role === "admin") {
+        setShouldShowForm(true)
+      } else {
+        setShouldRedirect(true)
+      }
+    })
+    .catch(error => console.error(`Error in fetch ${error.message}`))
+  }, [])
+
+  if(shouldRedirect) {
+    return <Redirect to="/parks" />
+  }
```

Comment:
this can stay here - if `shoudRedirect` is true, they will automatically be redirected - otherwise the container will return your form